clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -872,6 +875,11 @@ public void subscribe(Pattern pattern, ConsumerRebalanceListener listener) {
     public void unsubscribe() {
         acquire();
         try {
+            if (!this.subscriptions.allConsumed().isEmpty() && this.autoCommitEnabled)
```

Comment:
Without the additional check one of the existing unit tests would fail ([this line](https://github.com/vahidhashemian/kafka/blob/f60fb52a702a6bcb854e9560548174fc5ad86730/clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java#L147)) I believe because there is no offset commit callback specified (which should be OK because there is no consumed offsets). What do you think?